### PR TITLE
Fix custom modules not being added in ZGM missions

### PR DESCRIPTION
**Pull request by mharis001**
**When merged this pull request will:**
- Fix custom modules not being added to modules tree in ZGM missions

### CODE REVIEW

**Review by neilzar — APPROVED** [review]

**Review by Kexanone — CHANGES_REQUESTED** [review]
Nope

<details>
<summary>Recorded Test</summary>

https://user-images.githubusercontent.com/18646707/104492843-818f1e00-55d4-11eb-93e8-710323723ef8.mp4

</details>

**Comment by mharis001** [comment]
Yeah, doesn't work on dedicated since the curator addons takes time and the tree isn't refreshed in the next frame. Looking at it now, I am unsure about if there is a good way to support ZGM missions.

**Comment by Kexanone** [comment]
Yeah, it's an issue that has troubled us a lot in the past.
It's far from perfect, but Achilles basically checked for the presence of ZGM and counted the entries in the module tree.
Adapting this approach to unscheduled environment might be the best approach:

````sqf
sleep 0.001;
// For Zeus Game Master missions: Wait until respawn was placed
if (!(missionNamespace getVariable ["BIS_moduleMPTypeGameMaster_init", false]) && {count allMissionObjects "ModuleMPTypeGameMaster_F" > 0}) then
{
    waitUntil {sleep 0.1; _moduleTreeCtrl tvCount [] > 1 || isNull findDisplay IDD_RSCDISPLAYCURATOR};
};
if (isNull findDisplay IDD_RSCDISPLAYCURATOR) exitWith {};

// wait for the next frame
sleep 0.001;
[Achilles_fnc_onModuleTreeLoad, []] call CBA_fnc_directCall;
````

**Comment by Kexanone** [comment]
If I remember correctly `BIS_moduleMPTypeGameMaster_init` gets set to `true` by the ZGM module when the respawn is placed, but due to race conditions, you still have to check the number of entries in the module tree.

**Comment by Kexanone** [comment]
I'm also not sure if both "delay by one frame with `sleep 0.001`" are really necessary. I might have just tested everything possible and left some unnecessary stuff around.

File: addons/editor/functions/fnc_handleLoad.sqf [hunk 1]
```diff
@@ -128,10 +128,19 @@ GVAR(iconsVisible) = true;
...
         params ["_display"];
 
-        isNull _display || {entities "ModuleMPTypeGameMaster_F" isEqualTo []} || {GETMVAR(BIS_moduleMPTypeGameMaster_init,false)}
+        isNull _display
+        || {entities "ModuleMPTypeGameMaster_F" isEqualTo []}
+        || {
+            GETMVAR(BIS_moduleMPTypeGameMaster_init,false)
+            && {
+                (_display displayCtrl IDC_RSCDISPLAYCURATOR_CREATE_MODULES) tvCount [] > 1
+                || {"CuratorModeratorRights" call BIS_fnc_getParamValue < 1 && {player != BIS_curatorUnit}}
```
Comment (Kexanone) [inline addons/editor/functions/fnc_handleLoad.sqf:141]: Checking for an empty module tree (_i.e._ `tvCount` returns `0`) should already be enough for moderators without being too mission specific.

**Comment by Kexanone** [comment]
I can confirm that it works now for dedicated servers too.

File: addons/editor/functions/fnc_handleLoad.sqf (hunk 1 above)
Comment (mharis001) [reply addons/editor/functions/fnc_handleLoad.sqf:141]: This line is for when moderators have no addons and can only spectate or edit existing objects. In that case, the tree is always empty.

Comment (Kexanone) [reply addons/editor/functions/fnc_handleLoad.sqf:141]: Nvm, leave it as it is. My suggestion doesn't even work.

**Review by Kexanone — APPROVED** [review]

**Pull request merged**